src/qt/pivx/send.cpp

```diff
@@ -439,27 +439,32 @@ void SendWidget::updateEntryLabels(QList<SendCoinsRecipient> recipients)
     }
 }
 
-
 void SendWidget::onChangeAddressClicked()
 {
     showHideOp(true);
     SendChangeAddressDialog* dialog = new SendChangeAddressDialog(window, walletModel);
     if (!boost::get<CNoDestination>(&CoinControlDialog::coinControl->destChange)) {
         dialog->setAddress(QString::fromStdString(EncodeDestination(CoinControlDialog::coinControl->destChange)));
     }
-    if (openDialogWithOpaqueBackgroundY(dialog, window, 3, 5)) {
-        CTxDestination dest = DecodeDestination(dialog->getAddress().toStdString());
-        // Ask if it's what the user really wants
-        if (!walletModel->isMine(dest) &&
-            !ask(tr("Warning!"), tr("The change address doesn't belong to this wallet.\n\nDo you want to continue?"))) {
+
+    CTxDestination destChange = (openDialogWithOpaqueBackgroundY(dialog, window, 3, 5) ?
+                                 dialog->getDestination() : CNoDestination());
+
+    if (boost::get<CNoDestination>(&destChange)) {
```

Comment:
better to use `IsValidDestination` instead.

Reply:
:+1: Refactored in a separate commit, along with the other few instances found in the code.